src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,32 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    public boolean equals(Rectangle rectangle) {
+        if (rectangle == null) {
+            return false;
+        }
+        if (rectangle == this) {
+            return true;
+        }
+        if (rectangle.getClass().equals(Rectangle.class)) {
+            Rectangle compareRectangle = (Rectangle) rectangle;
+            return (this.width == compareRectangle.getWidth()
+                    || this.width.equals(compareRectangle.getWidth()))
+                    && (this.length == compareRectangle.getLength()
+                    || this.length.equals(compareRectangle.getLength()))
+                    && (this.color == compareRectangle.getColor()
+                    || this.color.equals(compareRectangle.getColor()));
```

Comment:
let's not use `this`

and what if `width` is null but `compareRectangle.getWidth()` is not null?